operator/pkg/name/name.go

```diff
@@ -49,8 +49,15 @@ const (
 	IngressComponentName ComponentName = "IngressGateways"
 	EgressComponentName  ComponentName = "EgressGateways"
 
-	// Addon components
-	AddonComponentName ComponentName = "Addon"
+	// Addon root component
+	AddonComponentName ComponentName = "AddonComponents"
+
+	// Legacy addon components
+	PrometheusComponentName ComponentName = "Prometheus"
+	KialiComponentName      ComponentName = "Kiali"
+	GrafanaComponentName    ComponentName = "Grafana"
+	TracingComponentName    ComponentName = "Tracing"
+	CoreDNSComponentName    ComponentName = "Istiocoredns"
```

Comment:
Remove these per discussion.

Reply:
Will we be able to enable them in some other way?